### DISCUSSION TITLE
Why have a custom sqrt implementation?

### DISCUSSION THREAD

**Discussion by JavaDerg** [mooman219/fontdue#84] [Q&A] [answered]
Hey, I've been looking trough the code earlier to try understand how the rasterization process works,
while doing so I found this file: https://github.com/mooman219/fontdue/blob/master/src/platform/float/sqrt.rs

Can someone explain me what is the reasoning behind having your own sqrt implementation over the one provided by rust?

**Comment by mooman219** [comment, marked as answer]
Some platforms that fontdue compiles for don't support the standard library, so f32::sqrt isn't available.

**Reply by JavaDerg** [reply]
Thanks for answering ^^
